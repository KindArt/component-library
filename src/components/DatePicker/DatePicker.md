Standard:

```tsx padded
<DatePicker label="Label" date={new Date()} />
```

Disabled:

```tsx padded
<DatePicker label="Label" disabled />
```
