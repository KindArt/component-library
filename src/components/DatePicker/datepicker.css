.DatePicker {
  width: 100%;
}

.customClass {
  display: flex;
  background-color: red;
}
