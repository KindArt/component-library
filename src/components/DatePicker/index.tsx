import React, { FC, useState, useEffect } from 'react';
import { enGB } from 'date-fns/locale';
import { DatePicker as CustomDatePicker } from 'react-nice-dates';
import { Input } from '../';
import 'react-nice-dates/build/style.css';

export interface IDatePickerProps {
  date?: Date;
  disabled?: boolean;
  label?: string;
  dateFormat?: string;
  locale?: string;
}

const DatePicker: FC<IDatePickerProps> = ({ date, disabled, label, dateFormat = 'DD/MM/YYYY', locale }) => {
  const [selectedDate, setDate] = useState<Date | null>(null);

  useEffect(() => {
    const newDate = date || null;
    setDate(newDate);
  }, [date]);

  const formattedDate = selectedDate || undefined;
  return (
    <CustomDatePicker date={formattedDate} onDateChange={setDate} locale={enGB}>
      {({ inputProps, focused }: any) => <Input label={label} disabled={disabled} {...inputProps} />}
    </CustomDatePicker>
  );
};

export default DatePicker;
