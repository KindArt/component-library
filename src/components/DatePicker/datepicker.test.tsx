import React from 'react';
import { render } from '@testing-library/react';
import DatePicker, { IDatePickerProps } from './';

describe('<DatePicker />', () => {
  const getDatePicker = (props?: IDatePickerProps) => {
    const utils = render(<DatePicker {...props} />);
    return { ...utils };
  };

  it('should render', () => {
    const { container } = getDatePicker();
    expect(container).toMatchSnapshot();
  });

  describe('Visuals', () => {
    it('should render a label when one is passed', () => {
      const newProps = { label: 'label' };
      const { queryByTestId } = getDatePicker(newProps);
      const label = queryByTestId('Label-wrapper');
      expect(label).toBeTruthy();
    });
  });
});
