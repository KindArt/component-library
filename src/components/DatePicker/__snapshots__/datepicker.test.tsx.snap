// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> should render 1`] = `
<div>
  <div
    class="nice-dates"
  >
    <div
      class="sc-AxhCb cdpuSL"
      data-testid="Input-wrapper"
    >
      <div
        class="sc-AxhUy A-DoCV"
      >
        <input
          class="sc-AxheI fvWmuM"
          data-testid="Input-input"
          placeholder="dd/mm/yyyy"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="nice-dates-popover"
    >
      <div>
        <div
          class="nice-dates-navigation"
        >
          <a
            class="nice-dates-navigation_previous"
          />
          <span
            class="nice-dates-navigation_current"
          >
            June
          </span>
          <a
            class="nice-dates-navigation_next"
          />
        </div>
        <div
          class="nice-dates-week-header"
        >
          <span
            class="nice-dates-week-header_day"
          >
            Mon
          </span>
          <span
            class="nice-dates-week-header_day"
          >
            Tue
          </span>
          <span
            class="nice-dates-week-header_day"
          >
            Wed
          </span>
          <span
            class="nice-dates-week-header_day"
          >
            Thu
          </span>
          <span
            class="nice-dates-week-header_day"
          >
            Fri
          </span>
          <span
            class="nice-dates-week-header_day"
          >
            Sat
          </span>
          <span
            class="nice-dates-week-header_day"
          >
            Sun
          </span>
        </div>
        <div
          class="nice-dates-grid"
          style="height: 6px;"
        >
          <div
            class="nice-dates-grid_container -origin-top"
            style="transform: translate3d(0, 0px, 0); transition-duration: 500ms;"
          >
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_month"
              >
                Jun
              </span>
              <span
                class="nice-dates-day_date"
              >
                1
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                2
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                3
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                4
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                5
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                6
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                7
              </span>
            </span>
            <span
              class="nice-dates-day -today"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                8
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                9
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                10
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                11
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                12
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                13
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                14
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                15
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                16
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                17
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                18
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                19
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                20
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                21
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                22
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                23
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                24
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                25
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                26
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                27
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                28
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                29
              </span>
            </span>
            <span
              class="nice-dates-day"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                30
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_month"
              >
                Jul
              </span>
              <span
                class="nice-dates-day_date"
              >
                1
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                2
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                3
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                4
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                5
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                6
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                7
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                8
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                9
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                10
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                11
              </span>
            </span>
            <span
              class="nice-dates-day -outside"
              style="height: 1px;"
            >
              <span
                class="nice-dates-day_date"
              >
                12
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
